src/DBConnection.hpp

```diff
@@ -1,14 +1,30 @@
 #pragma once
 
+#include <string>
+
+struct Info
+{
+	Info() : serverVersion(""), protocolVersion(0) {};
+	std::string serverVersion;
+	int protocolVersion;
+};
+
 class IDBConnection
 {
 public:
+	virtual ~IDBConnection() = default;
+
 	virtual int query() const = 0;
 };
 
 class MySqlDBConnection : public IDBConnection
 {
 public:
+	explicit MySqlDBConnection(std::string serverVersion, int protocolVersion);
+
 	int query() const override;
-	int advancedQuery() const;
+	const Info& advancedQuery() const;
```

Comment:
@Morheit , what you expect to win by returning ref? Another question, can I change fields of returned const ref?